### ISSUE TITLE
HTTP/3 support

### ISSUE THREAD

**Issue by bt90** [binwiederhier/ntfy#436]
HTTP/3 is finally standardized and it would probably a good fit for ntfys HTTPS endpoint as it allows for[ connection migration](https://http3-explained.haxx.se/en/quic/quic-connections):

> connections can thus migrate between IP addresses and network interfaces in ways TCP never could. For instance, migration allows an in-progress download to move from a cellular network connection to a faster wifi connection when the user moves their device into a location offering wifi. Similarly, the download can proceed over the cellular connection if wifi becomes unavailable.

As caddy enables HTTP/3 by default now, some users might already run a HTTP/3 capable ntfy server. For them it would be enough to add support for it in the Android and iOS clients.

https://github.com/lucas-clemente/quic-go

**Comment by binwiederhier** [comment]
I like the idea, and I've used this library in an experimental project 3 years ago (https://github.com/binwiederhier/natter). However, there are many other competing tickets, so this will likely sit for a while.

**Comment by bt90** [comment]
Yeah, perfectly fine. Without [WebTransport](https://www.w3.org/TR/webtransport/) support, it would only be useful for stream subscribers anyway.
